org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/utils/XMLBuilder.java

```diff
@@ -336,6 +339,32 @@ public void trimFinalNewlines() {
 		}
 	}
 
+	/**
+	 * Returns <code>str</code> with the trailing spaces from each line
+	 * removed
+	 *
+	 * @param str the String
+	 * @return <code>str</code> with the trailing spaces from each line
+	 * removed
+	 */
+	private String trimTrailingSpacesEachLine(String str) {

```

Comment:
set as static to show this method doesnt have some dependencies to some fields of this class.